enumerable.rb

```diff
@@ -0,0 +1,106 @@
+module Enumerable
+  def my_each
+    return enum_for unless block_given?
+
+    arr = is_a?(Range) ? to_a : self
+    i = 0
+    while i < arr.size
+      yield(arr[i])
+      i += 1
+    end
+    self
+  end
+
+  def my_each_with_index
+    return enum_for unless block_given?
+
+    arr = is_a?(Range) ? to_a : self
+    i = 0
+    while i < arr.size
+      yield(arr[i], i)
+      i += 1
+    end
+    self
+  end
+
+  def my_count
+    return length unless block_given?
+
+    arr = is_a?(Range) ? to_a : self
+    count = 0
+    i = 0
+    while i < arr.size
+      count += 1 if yield(arr[i])
+      i += 1
+    end
+    count
+  end
```

Comment:
* [ ] should have working `my_count` identical and returns the same thing as ruby's [`count`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-count). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_count.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))

#### my_count

- [ ] my_count counts the number of items in enum that are equal to the item if an argument is given
```ruby
[5, 4, 1].my_count(4) == [5, 4, 1].count(4) => 1
```
- Yours is giving an ArgumentError
![image](https://user-images.githubusercontent.com/52209067/79677237-475f3000-81ac-11ea-85fc-480d19f34353.png)

- Real count
![image](https://user-images.githubusercontent.com/52209067/77696378-e2306a00-6f72-11ea-9310-0bb30ab9c94d.png)

Reply:
Fixed!!
Thank you for the observation.